templates/toil-cluster-role-and-policy.yaml

```diff
@@ -0,0 +1,111 @@
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - !Ref ClusterPolicy
+  ClusterInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      InstanceProfileName: !Ref ClusterRole
+      Path: "/"
+      Roles:
+        - !Ref ClusterRole
+  ClusterPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Minimal permissions for toil instances.
+      ManagedPolicyName: !Join
+        - "-"
+        -
+          - !Ref ClusterName
+          - "-policy"
+      Path: "/"
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - ec2:CancelSpotInstanceRequests
+              - ec2:CreateSecurityGroup
+              - ec2:CreateTags
+              - ec2:DeleteSecurityGroup
+              - ec2:DescribeAvailabilityZones
+              - ec2:DescribeImages
+              - ec2:DescribeInstances
+              - ec2:DescribeInstanceStatus
+              - ec2:DescribeKeyPairs
+              - ec2:DescribeSecurityGroups
+              - ec2:DescribeSpotInstanceRequests
+              - ec2:DescribeSpotPriceHistory
+              - ec2:DescribeVolumes
+              - ec2:RequestSpotInstances
+              - ec2:RunInstances
+              - ec2:StartInstances
+              - ec2:StopInstances
+              - ec2:TerminateInstances
+              - iam:PassRole
+            Resource: "*"
+          - Effect: Allow
+            Action:
+              - s3:AbortMultipartUpload
+              - s3:CreateBucket
+              - s3:DeleteBucket
+              - s3:DeleteObject
+              - s3:DeleteObjectVersion
+              - s3:GetBucketLocation
+              - s3:GetBucketVersioning
+              - s3:GetObject
+              - s3:GetObjectVersion
+              - s3:GetObjectVersionTagging
+              - s3:HeadBucket
+              - s3:ListAllMyBuckets
+              - s3:ListBucket
+              - s3:ListBucketMultipartUploads
+              - s3:ListBucketVersions
+              - s3:ListMultipartUploadParts
+              - s3:PutBucketTagging
+              - s3:PutBucketVersioning
+              - s3:PutObject
+              - s3:PutObjectTagging
+              - s3:PutObjectVersionTagging
+              - s3:ReplicateTags
+            Resource: !Join
+              - ""
+              -
+                - "arn:aws:s3:::"
+                - !Ref ClusterName
+                - "*"
+          - Effect: Allow
+            Action:
+              - sdb:*
+            Resource: !Join
+              - ""
+              -
+                - "arn:aws:sdb:"
+                - !Ref "AWS::Region"
+                - ":"
+                - !Ref "AWS::AccountId"
+                - ":domain/"
+                - !Ref ClusterName
+                - "*"
```

Comment:
@zaro0508 Is there a more elegant/terse way to concatenate all this than using this `!Join` function?

Reply:
You can use `!Sub` docs here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html

Examples in the template to provision EC2 instances: https://github.com/Sage-Bionetworks/aws-infra/blob/master/templates/managed-ec2-v9.j2

Reply:
Thanks, that's much nicer. Changes in commit 6b0ce30.